core/threads.go

```diff
@@ -80,29 +102,33 @@ func (t *Textile) AddThread(sk libp2pc.PrivKey, conf AddThreadConfig) (*Thread,
 		set[m] = struct{}{}
 	}
 
-	threadModel := &repo.Thread{
+	model := &pb.Thread{
 		Id:        id.Pretty(),
 		Key:       conf.Key,
-		PrivKey:   skb,
+		Sk:        skb,
 		Name:      strings.TrimSpace(conf.Name),
-		Schema:    conf.Schema.B58String(),
-		Initiator: conf.Initiator,
+		Schema:    schema,
+		Initiator: initiator,
 		Type:      conf.Type,
 		Sharing:   conf.Sharing,
 		Members:   members,
-		State:     repo.ThreadLoaded,
+		State:     pb.Thread_Loaded,
 	}
-	if err := t.datastore.Threads().Add(threadModel); err != nil {
+	if err := t.datastore.Threads().Add(model); err != nil {
+		if conf.Force && repo.ConflictError(err) && strings.Contains(err.Error(), ".key") {
+			conf.Key = incrementKey(conf.Key)
```

Comment:
If `Force` is true in the thread config, force increment the thread key (adds "_1", "_2", "_3")... used when joining from an invite instead of just creating a new key.